conda_forge_tick/migrators.xsh

```diff
@@ -382,6 +382,15 @@ class Noarch(Migrator):
     compiler_pat = re.compile('.*_compiler_stub')
     sel_pat = re.compile('(.+?)\s*(#.*)?\[([^\[\]]+)\](?(2)[^\(\)]*)$')
     unallowed_reqs = ['toolchain', 'gcc', 'cython', 'clangdev']
+    checklist = ['No compiled extensions',
```

Comment:
Not that this is an issue for this PR but we might add this to the linter St some point as something for the maintainers to check if they are submitting a novel recipe or making a change.